tests/unit/multiclass/KNN_unittest.cc

```diff
@@ -226,3 +227,63 @@ TEST(KNN, classify_multiple_kdtree)
 
 
 }
+
+
+class NearestNeighbourParameterizedTestFixture :public ::testing::TestWithParam<KNN_SOLVER> {
+protected:
+	virtual void SetUp(){
+		lab = SGVector<float64_t>({0.0, 0.0, 1.0, 1.0, 0.0});
+		feat = SGMatrix<float64_t>({{2.0}, {3.0}, {6.0}, {7.0}, {3.5}});
+
+		SGVector<index_t> train ({0, 1, 2, 3});
+		SGVector<index_t> test ({4});
+
+		labels = std::make_shared<MulticlassLabels>(lab);
+		features = std::make_shared<DenseFeatures<float64_t> >(feat);
+
+		features_train = view(features, train); //{2.0}, {3.0}, {6.0}, {7.0}
+		labels_train = view(labels, train);	 //0.0, 0.0, 1.0, 1.0
+		features_test = view(features, test);	 //{3.5}
+		
+		distance = std::make_shared<EuclideanDistance>();
+		dist = std::make_shared<EuclideanDistance>(features_train, features_test);
+	}
+
+	virtual void TearDown()
+	{
+	}
+
+	SGVector<float64_t> lab;
+	SGMatrix<float64_t> feat;
+
+	std::shared_ptr<MulticlassLabels> labels;
+	std::shared_ptr<MulticlassLabels> labels_train;
+	std::shared_ptr<DenseFeatures<float64_t>> features;
+	std::shared_ptr<DenseFeatures<float64_t>> features_train;
+	std::shared_ptr<DenseFeatures<float64_t>> features_test;
+	std::shared_ptr<EuclideanDistance> distance;
+	std::shared_ptr<EuclideanDistance> dist;
+
+	const int32_t k=3; 
+};
+
+TEST_P(NearestNeighbourParameterizedTestFixture, CheckNearestNeighbours) {
+    KNN_SOLVER solver = GetParam();
+    
+	auto knn=std::make_shared<KNN> (k, distance, labels_train, solver);
+	knn->train(features_train);
+	knn->put("distance", (std::shared_ptr<shogun::Distance>)dist);
+
+	SGMatrix<index_t> NN = knn->nearest_neighbors();
+	
+	EXPECT_DOUBLE_EQ(feat(0, NN(0,0)), 3.0);	 //3.0 is closest to 3.5
+	EXPECT_DOUBLE_EQ(feat(0, NN(1,0)), 2.0);	 //2.0 is second-closest to 3.5
+	EXPECT_DOUBLE_EQ(feat(0, NN(2,0)), 6.0);	 //6.0 is third-closest to 3.5
+}
+
+INSTANTIATE_TEST_CASE_P(
```

Comment:
does this pass the tests?
I think in addition to some unit testing, could you also make some simple plots on some 2d data. So we can see whether this really does meaningful stuff? Alternatively, you could test the results on a nontrivial but simple dataset against a reference lib. And also I think you could test whether the different solvers return the same values on simple nontrivial data?